react/features/toolbox/components/web/Toolbox.tsx

```diff
@@ -1427,44 +1458,34 @@ class Toolbox extends Component<IProps> {
                                 ariaControls = 'overflow-menu'
                                 isOpen = { _overflowMenuVisible }
                                 key = 'overflow-menu'
+                                onToolboxEscKey = { this._onEscKey }
                                 onVisibilityChange = { this._onSetOverflowVisible }
-                                showMobileReactions = {
-                                    _reactionsEnabled && (_isMobile || _isNarrowLayout)
-                                }>
-                                <ContextMenu
-                                    accessibilityLabel = { t(toolbarAccLabel) }
-                                    className = { classes.contextMenu }
-                                    hidden = { false }
-                                    id = 'overflow-context-menu'
-                                    inDrawer = { _overflowDrawer }
-                                    onKeyDown = { this._onEscKey }>
-                                    {overflowMenuButtons.reduce((acc, val) => {
-                                        if (acc.length) {
-                                            const prev = acc[acc.length - 1];
-                                            const group = prev[prev.length - 1].group;
-
-                                            if (group === val.group) {
-                                                prev.push(val);
-                                            } else {
-                                                acc.push([ val ]);
-                                            }
+                                showRaiseHandInReactionsMenu = { showRaiseHandInReactionsMenu }
+                                showReactionsMenu = { showReactionsInOverflowMenu }>
+                                {overflowMenuButtons.reduce((acc, val) => {
+                                    if (val.key === 'reactions' && showReactionsInOverflowMenu) {
```

Comment:
Why do these need special treatment now?

Reply:
If we need to show reactions in the overflow menu we filter them out of the array with buttons just for convenience because we have separate property anyway and we display them differently than the other buttons. In theory we can probably move this logic into OverflowMenuButton but we were doing this for the raise hand already anyway and also we do the grouping here and iterate over the array so I thought it is not a bad idea to also filter them here.